mapreduce/src/main/java/org/pentaho/hadoop/mapreduce/PentahoMapRunnable.java

```diff
@@ -346,13 +346,15 @@ public Exception getException() {
 
   public void setDebugStatus( Reporter reporter, String message ) {
     if ( debug ) {
-      log.logBasic( message );
+      log.setLogLevel(LogLevel.DEBUG);
+      log.logDebug( message );
       reporter.setStatus( message );
     }
   }
 
   private void setDebugStatus( String message ) {
     if ( debug ) {
```

Comment:
This is not how it should be. This should be:

```
if (log.isDebugEnabled()) {
  log.logDebug( message );
}
```

Do not use "if (debug)" because this is hardcoded and will never be enabled. 

Do not use "setLogLevel()" because you're overriding the system administrator's configuration.